### PR TITLE
Add Synology DSM settings to NFS Server section

### PR DESCRIPTION
**Pull request by matonb**
Changes proposed in this pull request:

- Documentation: Defines work arounds to Synology DSM 7 ACL restrictions and lack of fine grained user control of NFS shares.

I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/master/CONTRIBUTING.md): @matonb

This pull request needs review by: @nirs

### CODE REVIEW

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 1]
```diff
@@ -99,7 +100,6 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
 
      #>/etc/init.d/nfs-kernel-server restart 
 
-
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:102]: This change is unrelated, it is better not to mix unrelated changes.

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 2]
```diff
@@ -147,6 +147,37 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
       # systemctl start rpcbind 
       # systemctl start nfs-server
 
+#### Synology DSM 7
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:150]: Blank line after a heading makes the source easier to read.

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 3]
```diff
@@ -147,6 +147,37 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
...
+Synology DSM 7 adds / extends Access Control Lists (ACLs) on all shared folders, and removes `groupadd` and `useradd`...  
+To allow ovirt nodes to connect via NFS the ACL permissions and /etc/exports file need to be updated manually (*take backups before editing!*).
+
+1. Create the shared folder required as usual and allow you oVirt hosts IP's in the NFS permissions section
+
+2. Login to you Synology NAS via SSH (search the web for 'synology enable ssh access') as root or sudo to root once logged in
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:156]: you -> your?

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 4]
```diff
@@ -147,6 +147,37 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
...
+
+3. Carefully add kvm (id 36) to /etc/group and vdsm (id 36) to /etc/passwd and /etc/shadow, if you need an encrypted password the python command below will create one
+
+        # python3 -c 'import crypt,getpass;pw=getpass.getpass();print(crypt.crypt(pw) if (pw==getpass.getpass("Confirm: ")) else exit())'
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:160]: Is this really the best way to add a user/group in Synology NAS?

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 5]
```diff
@@ -147,6 +147,37 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
...
+
+4. Update the Synology ACLs to allow access to kvm:vdsM
+
+        # EXPORT_DIR=/volumeX/...
+
+        # synoacltool -set-owner "$EXPORT_DIR" group kvm:allow:rwxpdDaARWcCo:fd--
+        # synoacltool -add "$EXPORT_DIR" user:vdsm:allow:rwxpdDaARWcCo:fd--
+        # synoacltool -add "$EXPORT_DIR" group:kvm:allow:rwxpdDaARWcCo:fd--
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:168]: From oVirt point of view, having 36:36 is good enough, there is no need
to add user and group. Does synoacltool support uid/guid or it requires user/group names?

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 6]
```diff
@@ -147,6 +147,37 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
...
+
+5.  Edit /etc/exports and replace the anonuid and anonguid values with 36, e.g.
+
+        /volume1/ovirt 1.1.1.1(rw,async,no_wdelay,crossmnt,insecure,no_root_squash,insecure_locks,sec=sys,anonuid=1025,anongid=100) 1.1.1.2(rw,async,no_wdelay,crossmnt,insecure,no_root_squash,insecure_locks,sec=sys,anonuid=1025,anongid=100)
+
+Would become
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:174]: I think the values before the change are not needed. Showing example line with anonuid=36,anongid=36
is better.

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 7]
```diff
@@ -147,6 +147,37 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
...
+
+        /volume1/ovirt 1.1.1.1(rw,async,no_wdelay,crossmnt,insecure,no_root_squash,insecure_locks,sec=sys,anonuid=36,anongid=36) 1.1.1.2(rw,async,no_wdelay,crossmnt,insecure,no_root_squash,insecure_locks,sec=sys,anonuid=36,anongid=36)
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:176]: Why not show simpler configuration in the examples?

    /volume1/ovirt *(rw,async,crossmnt,insecure,no_root_squash,insecure_locks,sec=sys,anonuid=36,anongid=36)

no_wdelay does not have any effect when async is used. Do we good reason to use crossmnt,insecure (with sec=sys) and no_root_squash with Synology DSM 7?

**Review by nirs — COMMENTED** [review]
Thanks @matonb, see inline comemnts.

File: source/develop/troubleshooting-nfs-storage-issues.md [hunk 8]
```diff
@@ -147,6 +147,37 @@ A new **nfs-check** script is now available to test whether an NFS export is rea
...
+
+1. Create the shared folder required as usual and allow you oVirt hosts IP's in the NFS permissions section
+
+2. Login to you Synology NAS via SSH (search the web for 'synology enable ssh access') as root or sudo to root once logged in
+
+3. Carefully add kvm (id 36) to /etc/group and vdsm (id 36) to /etc/passwd and /etc/shadow, if you need an encrypted password the python command below will create one
+
+        # python3 -c 'import crypt,getpass;pw=getpass.getpass();print(crypt.crypt(pw) if (pw==getpass.getpass("Confirm: ")) else exit())'
+
+4. Update the Synology ACLs to allow access to kvm:vdsM
```
Comment (nirs) [inline source/develop/troubleshooting-nfs-storage-issues.md:162]: vdsM -> vdsm

File: source/develop/troubleshooting-nfs-storage-issues.md (hunk 7 above)
Comment (matonb) [reply source/develop/troubleshooting-nfs-storage-issues.md:176]: Those are the options set by Synology, the only change I'm implying here is to the anonuid / anongid

File: source/develop/troubleshooting-nfs-storage-issues.md (hunk 4 above)
Comment (matonb) [reply source/develop/troubleshooting-nfs-storage-issues.md:160]: DSM 7 removes user / group add commands from the shell...  The built-in user management in the UI and the Synology replacement user add command (synouser I think) offer zero control of UID / GID numbers.

Actually I'd really just recommend avoiding Synology like the plague.

File: source/develop/troubleshooting-nfs-storage-issues.md (hunk 5 above)
Comment (matonb) [reply source/develop/troubleshooting-nfs-storage-issues.md:168]: Not sure I'll have a look, my setup was 'upgraded' and I'd already added the user and group with vanilla Linux commands on an earlier version.

File: source/develop/troubleshooting-nfs-storage-issues.md (hunk 6 above)
Comment (matonb) [reply source/develop/troubleshooting-nfs-storage-issues.md:174]: See reply below

File: source/develop/troubleshooting-nfs-storage-issues.md (hunk 5 above)
Comment (matonb) [reply source/develop/troubleshooting-nfs-storage-issues.md:168]: Unfortunately yes, it must be doing a lookup...

```
synoacltool -set-owner "$EXPORT_DIR" group 36:allow:rwxpdDaARWcCo:fd--
(synoacltool.c, 790)No such Group
```

File: source/develop/troubleshooting-nfs-storage-issues.md (hunk 2 above)
Comment (matonb) [reply source/develop/troubleshooting-nfs-storage-issues.md:151]: Added

File: source/develop/troubleshooting-nfs-storage-issues.md (hunk 3 above)
Comment (matonb) [reply source/develop/troubleshooting-nfs-storage-issues.md:158]: Well spotted

**Comment by nirs** [comment]
@matonb thanks, I'll review again next week.

**Comment by matonb** [comment]
@nirs, rebased and pushed again

**Review by nirs — APPROVED** [review]
Thanks!

**Comment by nirs** [comment]
@sandrobonazzola can you merge?

**Review by sandrobonazzola — APPROVED** [review]
